Invalid IPv6 format raises a panic

**Describe the bug**
If an IPv6 with the incorrect format is used a panic will be raised by the CLI.

**To Reproduce**
With the following `dnsconfig.js`:
```js
var REG = ...
var DSP = ...

D("example.com", REG, DnsProvider(DSP),
    AAAA("@", "2a00:1450:4003:809:1")
)
```

Running `dnscontrol preview` will panic with the following error:

```
panic: runtime error: slice bounds out of range [:16] with capacity 0

goroutine 1 [running]:
github.com/DNSControl/dnscontrol/v4/models.(*RecordConfig).ToRR(0x4d20fb2e008)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/models/record.go:401 +0x12ea
github.com/DNSControl/dnscontrol/v4/models.(*RecordConfig).zoneFileQuoted(0x24db340?)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/models/target.go:92 +0x6c
github.com/DNSControl/dnscontrol/v4/models.(*RecordConfig).GetTargetCombined(0x4d20fb2e008)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/models/target.go:79 +0x2f3
github.com/DNSControl/dnscontrol/v4/models.(*RecordConfig).ToComparableNoTTL(0x4d20fb2e008)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/models/record.go:357 +0x313
github.com/DNSControl/dnscontrol/v4/pkg/normalize.checkDuplicates({0x4d20fb19220, 0x2, 0x2a010dc?})
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/pkg/normalize/validate.go:674 +0x135
github.com/DNSControl/dnscontrol/v4/pkg/normalize.ValidateAndNormalizeConfig(0x4d20fb0ac60)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/pkg/normalize/validate.go:567 +0x253e
github.com/DNSControl/dnscontrol/v4/commands.prun({{{{0x29c3a61, 0xc}, {0x0, 0x0}, 0x0, {0x4584920, 0x0, 0x0}}, {0x0, 0x0}}, ...}, ...)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/commands/ppreviewPush.go:249 +0x429
github.com/DNSControl/dnscontrol/v4/commands.PPreview({{{{0x29c3a61, 0xc}, {0x0, 0x0}, 0x0, {0x4584920, 0x0, 0x0}}, {0x0, 0x0}}, ...})
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/commands/ppreviewPush.go:188 +0x85
github.com/DNSControl/dnscontrol/v4/commands.init.init.func8.func19({0x4d20f392588?, 0x2b42750?}, 0x4d20f4bd530?)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/commands/ppreviewPush.go:51 +0x59
github.com/urfave/cli/v3.(*Command).run(0x4d20f392588, {0x2b42750, 0x4d20f4bd530}, {0x4d20f4fb910, 0x1, 0x1})
        /home/jdobonb/.local/share/go/pkg/mod/github.com/urfave/cli/v3@v3.9.0/command_run.go:365 +0x2f19
github.com/urfave/cli/v3.(*Command).run(0x4d20f2942c8, {0x2b42750, 0x4d20f702de0}, {0x4d20f070060, 0x2, 0x2})
        /home/jdobonb/.local/share/go/pkg/mod/github.com/urfave/cli/v3@v3.9.0/command_run.go:299 +0x2745
github.com/urfave/cli/v3.(*Command).Run(...)
        /home/jdobonb/.local/share/go/pkg/mod/github.com/urfave/cli/v3@v3.9.0/command_run.go:93
github.com/DNSControl/dnscontrol/v4/commands.Run({0x4d20f726bc0, 0x1a})
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/commands/commands.go:128 +0x4d4
main.main()
        /home/jdobonb/.local/share/go/pkg/mod/github.com/!d!n!s!control/dnscontrol/v4@v4.41.0/main.go:24 +0xaa
```

**Expected behavior**
The CLI should show a proper error message.

--- a/models/dns_test.go
+++ b/models/dns_test.go
@@ -49,6 +49,20 @@ func TestRR(t *testing.T) {
 	}
 }
 
+func TestRRInvalidIPNoPanic(t *testing.T) {
+	// A malformed AAAA target (e.g. an incomplete IPv6 address) leaves the
+	// stored target unparseable. ToRR must not panic while building the RR so
+	// that normalize validation can report the error to the user.
+	experiment := RecordConfig{
+		Type:     "AAAA",
+		Name:     "@",
+		NameFQDN: "example.com",
+		target:   "2a00:1450:4003:809:1",
+		TTL:      300,
+	}
+	experiment.ToRR()
+}
+
 func TestSvcbAutoHintsTargetCombined(t *testing.T) {
 	experiment := RecordConfig{
 		Type:        "HTTPS",

--- a/models/record.go
+++ b/models/record.go
@@ -393,12 +393,14 @@ func (rc *RecordConfig) ToRR() dnsv1.RR {
 	switch rdtype { // #rtype_variations
 	case dnsv1.TypeA:
 		addr := rc.GetTargetIP()
-		s := addr.AsSlice()
-		rr.(*dnsv1.A).A = s[0:4]
+		if s := addr.AsSlice(); len(s) == 4 {
+			rr.(*dnsv1.A).A = s
+		}
 	case dnsv1.TypeAAAA:
 		addr := rc.GetTargetIP()
-		s := addr.AsSlice()
-		rr.(*dnsv1.AAAA).AAAA = s[0:16]
+		if s := addr.AsSlice(); len(s) == 16 {
+			rr.(*dnsv1.AAAA).AAAA = s
+		}
 	case dnsv1.TypeCAA:
 		rr.(*dnsv1.CAA).Flag = rc.CaaFlag
 		rr.(*dnsv1.CAA).Tag = rc.CaaTag

